src/end-to-end.test.js

```diff
@@ -390,3 +399,90 @@ it("can retrieve the list of mounted graph files", async () => {
 
   expect(screen.queryByText("cactus.vg.xg")).toBeTruthy();
 });
+
+// uploads a cactus.vg file and renders the svg
+it("can accept uploaded files", async () => {
+  await waitForLoadEnd();
+
+  // Swap over to the custom files mode
+  await act(async () => {
+    let dropdown = document.getElementById("dataSourceSelect");
+    await userEvent.selectOptions(
+      screen.getByLabelText(/Data/i),
+      "custom"
+    );
+  });
+
+  // Find the select box's input
+  let trackSelectButton = screen.queryByTestId("TrackPickerButton");
+  expect(trackSelectButton).toBeTruthy();
+
+  // open track selection
+  await act(async () => {
+    userEvent.click(trackSelectButton);
+  });
+
+
+  // add a new track
+  await waitFor(() => {
+    fireEvent.click(screen.queryByTestId("track-add-button-component"));
+  });
+
+  await waitFor(() => {
+    // wait for picker type dropdown
+    fireEvent.keyDown(screen.queryByTestId('picker-type-select-component1').firstChild, {key: "ArrowDown"});
+  });
+
+  // select the upload option
+  await waitFor(() => screen.getByText("upload"));
+  fireEvent.click(screen.getByText("upload"));
+
+  await waitFor(() => screen.queryByTestId("file-select-component1"));
+
+  const fileUploader = screen.queryByTestId("file-select-component1");
+  const fakeFile = new File(["example_data"], "example.vg", { type: "text/vg" });
```

Comment:
I don't think `text/vg` is right; it should really be `application/octet-stream`, which is probably what the browser is probably going to see it as.